### PR TITLE
Optimized STAC API property extraction in data store and mosaic

### PR DESCRIPTION
**Pull request by aaime**
Couple of related changes to optimize the data transfer to STAC APIs:
* https://osgeo-org.atlassian.net/browse/GEOT-7277
* https://osgeo-org.atlassian.net/browse/GEOT-7279

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [x] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [x] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [ ] Bug fixes and small new features are presented as a single commit.
- [x] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

### CODE REVIEW

**Comment by aaime** [comment]
@turingtestfail could you review the first commit, related to the STAC API store?
@dromagnoli could you review the second commit, which is in image mosaic land?

File: modules/unsupported/stac-store/src/main/java/org/geotools/stac/store/STACFeatureTypeFactoryImpl.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                boolean isAbstract,
+                List<Filter> restrictions,
+                AttributeType superType,
+                InternationalString description) {
+            super(name, schema, defaultGeometry, isAbstract, restrictions, superType, description);
+        }
+
+        @Override
+        public AttributeDescriptor getDescriptor(String name) {
+            if (name.startsWith("assets/")) return super.getDescriptor("assets");
```
Comment (turingtestfail) [inline modules/unsupported/stac-store/src/main/java/org/geotools/stac/store/STACFeatureTypeFactoryImpl.java:60]: Replace assets/ with constant?

File: modules/unsupported/stac-store/src/main/java/org/geotools/stac/store/SearchQueryBuilder.java [hunk 2]
```diff
@@ -191,6 +202,16 @@ public Pair<SearchQuery, Filter> toSearchQuery(Query q, boolean strict) {
         return Pair.of(sq, post);
     }
 
+    /**
+     * Most properties are found under "properties" but a few are top level. They need different
+     * paths in the JSON structure.
+     */
+    private static String propertyToField(String n) {
+        return n.equals("geometry") || n.equals("assets") || n.startsWith("assets/")
```
Comment (turingtestfail) [inline modules/unsupported/stac-store/src/main/java/org/geotools/stac/store/SearchQueryBuilder.java:210]: Same thing, maybe make these constants?

File: modules/unsupported/stac-store/src/main/java/org/geotools/stac/store/SearchQueryBuilder.java [hunk 3]
```diff
@@ -177,12 +177,23 @@ public Pair<SearchQuery, Filter> toSearchQuery(Query q, boolean strict) {
...
                         nameList.stream()
-                                .map(n -> n.equals("geometry") ? n : "properties." + n)
+                                .map(n -> propertyToField(n))
+                                .map(n -> n.replace('/', '.'))
                                 .collect(Collectors.toList());
                 fields.add("type"); // some servers would even remove this field if not included!
                 fields.add("id"); // and this as well
-                fields.add(
-                        "-links"); // some servers like to return the links no matter what instead
+                fields.add("-bbox"); // no need for the top level bbox
```
Comment (turingtestfail) [inline modules/unsupported/stac-store/src/main/java/org/geotools/stac/store/SearchQueryBuilder.java:185]: Same thing, maybe make these constants?

Comment (aaime) [reply modules/unsupported/stac-store/src/main/java/org/geotools/stac/store/SearchQueryBuilder.java]: Done

File: modules/unsupported/stac-store/src/main/java/org/geotools/stac/store/SearchQueryBuilder.java (hunk 2 above)
Comment (aaime) [reply modules/unsupported/stac-store/src/main/java/org/geotools/stac/store/SearchQueryBuilder.java]: Done

**Pull request merged**